bigtable/client/table_admin.h

```diff
@@ -190,6 +192,25 @@ class TableAdmin {
   bool CheckConsistency(bigtable::TableId const& table_id,
                         bigtable::ConsistencyToken const& consistency_token);
 
+  /**
+   * Using a separate thread
+   *
+   * @param table_id the id of the table for which we want to check
+   *     consistency.
+   * @param consistency_token the consistency token of the table.
+   * @param polling_policy the policy applicable for asynchronous call.
+   * @return the consistency status for the table.
+   * @throws std::exception if the operation cannot be completed.
+   */
+  std::future<bool> WaitForConsistencyCheck(
+      bigtable::TableId const& table_id,
+      bigtable::ConsistencyToken const& consistency_token,
+      std::unique_ptr<bigtable::PollingPolicy> polling_policy) {
```

Comment:
I think maybe PollingPolicy be an attribute of the class instead of an argument?  Thoughts?

Reply:
Yes we can make PollingPolicy as a private attribute which will be created at runtime using the values of rpc_retry_policy and rpc_backup_policy. So the user will be unknown to this attribute and he will not need to deal with PollingPolicy objects.
Some questions come to my mind.
1. Are we planning to have more functionality in PollingPolicy, than the the combination of 
rpc_retry_policy and rpc_backoff_policy.
   If yes then we must have to pass PollingPolicy as a argument to the TableAdmin constructor.
   If no then we can keep this as a private member used internally for asynchronous processing.
Shall I start to work on this, to make PollingPolicy as a part of the TableAdmin class?

Reply:
We should pass this as an argument to the `TableAdmin` constructor, just like we do for other policies.